CHANGELOG.md

```diff
@@ -2,6 +2,92 @@
 
 This contains only the most important and/or user-facing changes; for a full changelog, see the commit history.
 
+## [2.0.0](https://github.com/ably/ably-js/tree/2.0.0) (date TBD)
+
+The 2.0.0 release introduces a number of new features and QoL improvements, including a new way to remove bloat and reduce the bundle size of your ably-js client, first-class support for Promises, a more idiomatic approach to using ably-js' React Hooks, enhancements to TypeScript typings, and more.
+
+Below is an overview of the major changes in this release.
+
+Please refer to the ably-js v2 [lib migration guide](./docs/migration-guides/v2/lib.md) and [React Hooks migration guide](./docs/migration-guides/v2/react-hooks.md) for the full details, including a list of all breaking changes and instructions on how to address them.
+
+### Bundle Size Reduction
+
+The default bundle size for the web has been reduced by 30% compared to v1 - from 231.35 KiB to 162.15 KiB. When calculated with gzip compression, the reduction is 28.5%, from 81.68 KiB to 58.48 KiB.
```

Comment:
Just to check — are these numbers up to date for latest v1 and v2 (especially after the removal of `xhr_streaming` in v2)?

Reply:
Good point!
Updated in https://github.com/ably/ably-js/pull/1656/commits/9b908a0942a29477ea75f702f8931f99c22f2593.
v1 actually increased a bit in size (those deprecation messages come at a cost), but v2 decreased thanks to removal of `xhr_streaming` and connection upgrade code

Reply:
Awesome! Feel free to merge 🙂